web/controllers/pair_controller.ex

```diff
@@ -24,13 +25,17 @@ defmodule Pairmotron.PairController do
   end
 
   def delete(conn, %{"year" => y, "week" => w}) do
-    {year, _} = y |> Integer.parse
-    {week, _} = w |> Integer.parse
-    fetch_pairs(year, week)
-      |> Enum.map(fn(p) -> Repo.delete! p end)
-    conn
-      |> put_flash(:info, "Repairified")
-      |> redirect(to: pair_path(conn, :show, year, week))
+    if conn.assigns.current_user.is_admin do
```

Comment:
Since there's no :id parameter in the :index route, canary yells that it's doing something unsafe because it is comparing against nil. Unfortunately, I have to do the manual style admin check here.